lib/mix/tasks/phx.gen.release.ex

```diff
@@ -0,0 +1,169 @@
+      ])
+    end
+
+    if docker? do
+      Mix.Phoenix.copy_from(paths(), "priv/templates/phx.gen.release", binding, [
+        {:eex, "Dockerfile.eex", "Dockerfile"},
+        {:eex, "dockerignore.eex", ".dockerignore"}
+      ])
+    end
+
+    File.chmod!("rel/overlays/bin/server", 0o700)
+    if ecto?, do: File.chmod!("rel/overlays/bin/migrate", 0o700)
+
+    Mix.shell().info("""
+
+    Your application is ready to be deployed in a release!
+
+        # To start your system
+        _build/dev/rel/#{app}/bin/#{app} start
+
+        # To start your system with the Phoenix server running
+        _build/dev/rel/#{app}/bin/server
+        #{ecto? && ecto_instructions(app)}
+    Once the release is running:
+
+        # To connect to it remotely
+        _build/dev/rel/#{app}/bin/#{app} remote
+
+        # To stop it gracefully (you may also send SIGINT/SIGTERM)
+        _build/dev/rel/#{app}/bin/#{app} stop
+
+    To list all commands:
+
+        _build/dev/rel/#{app}/bin/#{app}
+    """)
+
+    ecto? &&
+      post_install_instructions("config/runtime.exs", ~r/ECTO_IPV6/, """
+      [warn] Conditional IPV6 support missing from runtime configuration.
+
+      Add the following to your config/runtime.exs:
+
+          ipv6? = System.get_env("ECTO_IPV6") == "true"
+
+          config :#{app}, #{app_namespace}.Repo,
+            ...,
+            socket_options: if(ipv6?, do: [:inet6], else: [])
+      """)
+
+    post_install_instructions("config/runtime.exs", ~r/PHX_SERVER/, """
+    [warn] Conditional server startup is missing from runtime configuration.
+
+    Add the following to your config/runtime.exs:
+
+        server? = System.get_env("PHX_SERVER") == "true"
+
+        config :#{app}, #{app_namespace}.Endpoint,
+          ...,
+          server: server?
+    """)
+
+    post_install_instructions("config/runtime.exs", ~r/PHX_HOST/, """
+    [warn] Environment based URL export is missing from runtime configuration.
+
+    Add the following to your config/runtime.exs:
+
+        host = System.get_env("PHX_HOST") || "example.com"
+
+        config :#{app}, #{app_namespace}.Endpoint,
+          ...,
+          url: [host: host, port: 443]
+    """)
+  end
+
+  defp ecto_instructions(app) do
+    """
+
+        # To run migrations
+        _build/dev/rel/#{app}/bin/migrate
+    """
+  end
+
+  defp paths do
+    [".", :phoenix]
+  end
+
+  defp post_install_instructions(path, matching, msg) do
+    case File.read(path) do
+      {:ok, content} ->
+        unless content =~ matching, do: Mix.shell().info(msg)
+
+      {:error, _} ->
+        Mix.shell().info(msg)
+    end
+  end
+
+  def otp_vsn do
+    major = to_string(:erlang.system_info(:otp_release))
+    path = Path.join([:code.root_dir(), "releases", major, "OTP_VERSION"])
+
+    with {:ok, content} <- File.read(path),
+         {:ok, %Version{} = vsn} <- Version.parse(String.trim_trailing(content)) do
+      "#{vsn.major}.#{vsn.minor}.#{vsn.patch}"
```

Comment:
OTP versioning is... tricky. There is no 24.1.0, it is just 24.1. But i guess you could do this:

```suggestion
    with {:ok, content} <- File.read(path) do
      String.trim(content)
```